pkgs/os-specific/linux/bpftools/strip-binary-name.patch

```diff
@@ -0,0 +1,15 @@
+Strip path to the binary from prints.
+
+I see no sense in including the full path in outputs like bpftool --version
+Especially as argv[0] may not include it, based on calling via $PATH or not.
+--- a/tools/bpf/bpftool/main.c
++++ b/tools/bpf/bpftool/main.c
+@@ -443 +443,7 @@
+-	bin_name = argv[0];
++	/* Strip the path if any. */
++	const char *bin_name_slash = strrchr(argv[0], '/');
```

Comment:
Another bug is assuming that there is `argv[0]` set here.

Reply:
We simplify the implementation and fix this by using `program_invocation_short_name` here.  It's non-standard but implemented by at least Glibc and Musl.

Reply:
That doesn't address this case of `argv[0] == NULL`.  We'd also need to add `-D_GNU_SOURCE` or similar.

Reply:
Upstream accepted a simpler approach of hard-coding the name instead, but I suppose we don't need to care and we'll just get it in some future update &ndash; at which point this patch will conflict and get dropped.